PyEMD/checks.py

```diff
@@ -0,0 +1,117 @@
+from PyEMD import EMD, EEMD, CEEMDAN
+import numpy as np
+from scipy import stats
+import math
+from scipy.signal import argrelextrema
+
+#helper function: normalize data
+def normalize(data):
+    return (data - data.mean())/(data.max() - data.min())
+
+#helper function: Find mean period of an IMF
+def mean_period(data):
+    return len(data)/len(argrelextrema(data, np.greater)[0])
+
+#helper function: find energy of signal/IMF
+def energy(data):
+    return sum(pow(data, 2))
+
+#helper function: find IMF significance in 'a priori' test
+def significance_apriori(energy_density, T, N, alpha):
+    k = abs(stats.norm.ppf((1-alpha)/2))
+    upper_limit = -math.log(T) + ( k * (math.sqrt(2/N) * math.exp(math.log(T)/2)) )
+    lower_limit = -math.log(T) - ( k * (math.sqrt(2/N) * math.exp(math.log(T)/2)) )
+
+    return not (lower_limit <= energy_density <= upper_limit)
+
+
+#helper function: find significance in 'a posteriori' test
+def significance_aposteriori(scaled_energy_density, T, N, alpha):
+    k = abs(stats.norm.ppf((1-alpha)/2))
+    lower_limit = -math.log(T) + ( k * (math.sqrt(2/N) * math.exp(math.log(T)/2)) )
+
+    return not (scaled_energy_density <= lower_limit)
+
+
+def whitenoise_check(data: np.ndarray, method: str='EMD',  test: str='aposteriori', rescaling_imf: int=1, alpha: float= 0.95, **kwargs):
+    """Returns a dictionary with IMF number(index starts at 1) as key; 0 value means IMF has noise in it and is not statistically significant; 1 value means IMF has some information and is statistically significant
```

Comment:
It isn't enforced yet (might be in a few hours) but the line length for the code is 120 characters. Please also refer to PEP on docstring and specifically on mulit-line docstrings ([PEP257](https://www.python.org/dev/peps/pep-0257/#id17)). The first line should be short summary followed by actual description. In your case, the first line would be something like `"Whitenoise statistical significance test"`.

Reply:
Made the changes and checked with pep257 tool, found no errors.